sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py

```diff
@@ -114,3 +180,106 @@ def _get_app(self):
+                client_id=self._auth_record.client_id,
+                authority=self._auth_record.authority,
+                tenant_id=tenant_id,
+                **kwargs
+            )
+        else:
+            super(InteractiveCredential, self).__init__(**kwargs)
+
+    def get_token(self, *scopes, **kwargs):
+        # type: (*str, **Any) -> AccessToken
+        """Request an access token for `scopes`.
+
+        .. note:: This method is called by Azure SDK clients. It isn't intended for use in application code.
+
+        :param str scopes: desired scopes for the access token. This method requires at least one scope.
+        :rtype: :class:`azure.core.credentials.AccessToken`
+        :raises CredentialUnavailableError: the credential is unable to attempt authentication because it lacks
+          required data, state, or platform support
+        :raises ~azure.core.exceptions.ClientAuthenticationError: authentication failed. The error's ``message``
+          attribute gives a reason.
+        :raises AuthenticationRequiredError: user interaction is necessary to acquire a token, and the credential is
+          configured not to begin this automatically. Call :func:`authenticate` to begin interactive authentication.
+        """
+        if not scopes:
+            raise ValueError("'get_token' requires at least one scope")
+
+        allow_prompt = kwargs.pop("_allow_prompt", not self._disable_automatic_authentication)
+        try:
+            return self._acquire_token_silent(*scopes, **kwargs)
+        except AuthenticationRequiredError:
+            if not allow_prompt:
+                raise
+
+        # silent authentication failed -> authenticate interactively
+        now = int(time.time())
+
+        result = self._request_token(*scopes, **kwargs)
+        if "access_token" not in result:
+            message = "Authentication failed: {}".format(result.get("error_description") or result.get("error"))
+            raise ClientAuthenticationError(message=message)
+
+        # this may be the first authentication, or the user may have authenticated a different identity
+        self._auth_record = _build_auth_record(result)
+
+        return AccessToken(result["access_token"], now + int(result["expires_in"]))
+
+    def authenticate(self, **kwargs):
+        # type: (**Any) -> AuthenticationRecord
+        """Interactively authenticate a user.
+
+        :keyword Sequence[str] scopes: optional scopes to request during authentication, such as those provided by
+          :func:`AuthenticationRequiredError.scopes`. If provided, successful authentication will cache an access token
+          for these scopes.
+        :rtype: ~azure.identity.AuthenticationRecord
+        :raises ~azure.core.exceptions.ClientAuthenticationError: authentication failed. The error's ``message``
+          attribute gives a reason.
+        """
+
+        scopes = kwargs.pop("scopes", None) or ("https://management.azure.com/.default",)
```

Comment:
After incorporating #11050, I plan to use the authority the credential was initialized with to choose a default scope.